ensure kwargs are always in sync across the whole application

pngwn opened the pull request:
## Description

Closes #7786. Closes #7895.

The issue was not present in dev mode but _was_ present in the built files. So we weren't noticing this behaviour day to day in development.

The root cause, seems to be that the list of props that we were using to listen for changes wasn't the same in dev + build. I'm not 100% why really, its a little confusing but I have confirmed that the approach in this PR works for all modes.

I still need to look at the test, but I need to create a failing test from main first. Will update shortly but the code is good for review.

To test you can use the `blocks_flipper` demo. This does not behave itself on main but works correctly in dev. You can also use the repro in #7895 as it is the same issue.

Be sure to build the frontend locally and visit port `7860` and not the dev port (9876). 

**Update**

I've update the test. The test was opening the accordion, clicking the slider and then checking that accordion text was on the page. The issue here is that for a brief moment after changing the slider value this is true, so sometimes this was passing (sometime it was failing too but we have retries on).

I've switched it out for a test that opens the accordion, changes the slider value then enters text into the textbox and waits for the server response + state update. This guarantees that the UI state has fully settled before asserting on the accordion contents.

This test passes on this branch but it fails consistently on main.

freddyaboulton reviewed (approved):
Can confirm this fixes the bug @pngwn ! Can this demo be added to 2e2 tests?

pngwn commented on js/app/src/RenderComponent.svelte:
```diff
@@ -28,9 +28,7 @@
 				//@ts-ignore
 				const instance = new _target(...args);
-				const props = Object.getOwnPropertyNames(instance).filter(
-					(s) => !s.startsWith("$")
```
In addition to the prop names, there are also some internal svelte things (methods mostly) that start with `$`. Was always a little hacky.

pngwn commented:
@freddyaboulton Which demo? The blocks flipper is the original test case hear and is in the e2e tests, or do you mean the streaming one?

freddyaboulton commented:
Oh weird I didn't know we already had a test! I guess `toBeVisible` is not the right playwright assertion to use? The test should not be passing right?

freddyaboulton commented:
Good to merge, we can take a look at the test later.

pngwn commented:
I updated that test so that it doesn't / would fail consistently on main. The assertion was okay, its just that the test was running faster than the UI could update. This is why it was flakey. Funnily enough the flake was the 'correct' behaviour.

> I've updated the test. The test was opening the accordion, clicking the slider and then checking that accordion text was on the page. The issue here is that for a brief moment after changing the slider value, this is true. So sometimes this was passing (sometimes it was failing too but we have retries on).
>
> I've switched it out for a test that opens the accordion, changes the slider value then enters text into the textbox and waits for the server response + state update. This guarantees that the UI state has fully settled before asserting on the accordion contents.
>
> This test passes on this branch but it fails consistently on main.

freddyaboulton commented:
Nice thanks @pngwn !
